anesthetic/weighted_pandas.py

```diff
@@ -2,31 +2,31 @@
 
 import numpy as np
 import pandas
-from warnings import warn
 from anesthetic.utils import compress_weights, channel_capacity, quantile
 
 
 class _WeightedObject(object):
     @property
     def weight(self):
         """Sample weights."""
-        if self._weight is None:
+        if self.index.nlevels == 1:
             return pandas.Series(index=self.index, data=1.)
         else:
-            return self._weight[self.index]
+            return pandas.Series(data=self.index.droplevel(0),
+                                 index=self.index)
+
+    @weight.setter
+    def weight(self, weight):
+        if weight is not None:
+            self.index = [self.index.get_level_values(0), weight]
+            self.index.set_names([None, 'weight'], inplace=True)
```

Comment:
You chose not to give a name to the iteration index. I agree that it is not as necessary from a user point of view to have a name displayed in the table. However, for our future selves looking at this code a name might be clearer than `0` and `None`. Maybe something as simple as `'#'` would already be clearer for us?